From 6eb3fa82ca8634aa2b79d5b74ef09cb1a2e5d3e5 Mon Sep 17 00:00:00 2001
From: Wolfgang Pupp <wolfgang.pupp@gmail.com>
Date: Tue, 23 Feb 2016 08:11:36 +0100
Subject: [PATCH] cygpath-conversion in npm-shell wrapper

Always call cygpath if the command is available and the node executable
is node.exe
---
 deps/npm/bin/npm | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/deps/npm/bin/npm b/deps/npm/bin/npm
index 0b37c82..2c9754b 100755
--- a/deps/npm/bin/npm
+++ b/deps/npm/bin/npm
@@ -3,11 +3,10 @@
 
 basedir=`dirname "$0"`
 
-case `uname` in
-    *CYGWIN*) basedir=`cygpath -w "$basedir"`;;
-esac
-
 if [ -x "$basedir/node.exe" ]; then
+  if [ -x "$(command -v cygpath)" ]; then
+    basedir=`cygpath -w "$basedir"`
+  fi
   "$basedir/node" "$basedir/../lib/node_modules/npm/bin/npm-cli.js" "$@"
 else
   node "$basedir/../lib/node_modules/npm/bin/npm-cli.js" "$@"
-- 
2.7.0

